bin/commitments_deprecater.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+require "cluster"
+require "services"
+require "deprecation_record"
+
+Services.mongo!
+
+# Takes (shared print) deprecation records, and tries to match them to commitments
+# so that those commitments can be deprecated.
+# Usage: CommitmentsDeprecater.new.run(<list_of_files>)
+
+class CommitmentsDeprecater
+  attr_reader :err, :log_path, :report
+
+  def initialize(verbose: false)
+    @verbose = verbose
+    @header_spec = ["organization", "ocn", "local_id", "deprecation_status"]
+  end
+
+  # Process a list of files.
+  def run(args)
+    args.each do |arg|
+      if File.file?(arg)
+        from_file(arg)
+      else
+        log "arg #{arg} not a file, skipping"
```

Comment:
Would we normally expect this? Is it OK just to get a file not found exception and terminate if given something that isn't a file?

Reply:
Would I expect that a piece of code which gets file paths from a human occasionally gets something that isn't a file path? Absolutely, I would. Especially if that human is giving paths that aren't even on their local machine via Tanka or whatever.
Is a runtime error preferred? I could go either way.